File: Adapter/Golang/MediaAdapter.go
package main

import (
	"fmt"
	"strings"
)

type MediaAdapter struct {
	Player IMediaPlayer
}

func (mediaAdapter *MediaAdapter) playMovie(filename string)  {
	filename_detail := strings.Split(strings.ToLower(filename), ".")
	count_arr := len(filename_detail)

	if count_arr < 2 {
		fmt.Println("File not valid")
		return
	}

	switch filename_detail[count_arr - 1] {
		case "mkv":
			mediaAdapter.Player.playMKV(filename)
			break

		case "mp4":
			mediaAdapter.Player.playMP4(filename)
			break
		
		default:
			fmt.Println("File type not supported")
	}
}

File: Adapter/Dart/MediaAdapter.dart
import 'IMediaPlayer.dart';

class MediaAdapter {
  IMediaPlayer? _mediaPlayer;

  MediaAdapter(IMediaPlayer mediaPlayer) {
    _mediaPlayer = mediaPlayer;
  }

  void playMovie(String filename) {
    final filename_detail = filename.split('.');
    final count_arr = filename_detail.length;

    if (count_arr < 2) {
      print('File not valid');
      return;
    }

    switch (filename_detail[count_arr - 1].toLowerCase()) {
      case "mkv":
        _mediaPlayer?.playMKV(filename);
        break;

      case "mp4":
        _mediaPlayer?.playMP4(filename);
        break;

      default:
        print('File type not supported');
        break;
    }
  }
}
